EventFilter/L1TRawToDigi/plugins/implementations_stage2/EtSumUnpacker_0x10010057.cc

```diff
@@ -0,0 +1,250 @@
+
+       LogDebug("L1T") << "MET: phi " << met.hwPhi() << " pT " << met.hwPt() << " bx " << bx;
+
+       res_->push_back(bx,met);
+
+       // MBHFPT1
+
+       l1t::EtSum mbp1 = l1t::EtSum();
+       mbp1.setHwPt( (raw_data>>28) & 0xf );
+       mbp1.setType( l1t::EtSum::kMinBiasHFP1 );
+
+       res_->push_back(bx, mbp1);
+
+       // ET ASYMM
+
+       l1t::EtSum etAsym = l1t::EtSum();
+       etAsym.setHwPt( (raw_data>>20) & 0xFF );
+       etAsym.setType( l1t::EtSum::kAsymEt );
+
+       res_->push_back(bx, etAsym);
+
+       // MHT 
+
+       raw_data = block.payload()[iFrame+3];
+
+       l1t::EtSum mht = l1t::EtSum();
+    
+       mht.setHwPt(raw_data & 0xFFF);
+       mht.setHwPhi((raw_data >> 12) & 0xFF);
+       mht.setType(l1t::EtSum::kMissingHt);       
+       mht.setP4( l1t::CaloTools::p4Demux(&mht) );
+
+       LogDebug("L1T") << "MHT: phi " << mht.hwPhi() << " pT " << mht.hwPt() << " bx " << bx;
+
+       res_->push_back(bx,mht);
+
+       // MBHFMT1
+
+       l1t::EtSum mbm1 = l1t::EtSum();
+       mbm1.setHwPt( (raw_data>>28) & 0xf );
+       mbm1.setType( l1t::EtSum::kMinBiasHFM1 );
+
+       res_->push_back(bx, mbm1);
+
+       // HT ASYMM
+
+       l1t::EtSum htAsym = l1t::EtSum();
+       htAsym.setHwPt( (raw_data>>20) & 0xFF );
+       htAsym.setType( l1t::EtSum::kAsymHt );
+
+       res_->push_back(bx, htAsym);
+
+       
+       //  MET (with HF)
+
+       raw_data = block.payload()[iFrame+4];
+
+       l1t::EtSum methf = l1t::EtSum();
+    
+       methf.setHwPt(raw_data & 0xFFF);
+       methf.setHwPhi((raw_data >> 12) & 0xFF);
+       methf.setType(l1t::EtSum::kMissingEtHF);       
+       methf.setP4( l1t::CaloTools::p4Demux(&methf) );
+
+       LogDebug("L1T") << "METHF: phi " << methf.hwPhi() << " pT " << methf.hwPt() << " bx " << bx;
+
+       res_->push_back(bx,methf);
+
+       // ETHF ASYMM
+
+       l1t::EtSum etHFAsym = l1t::EtSum();
+       etHFAsym.setHwPt( (raw_data>>20) & 0xFF );
+       etHFAsym.setType( l1t::EtSum::kAsymEtHF );
+
+       res_->push_back(bx, etHFAsym);
+
+       // CENTRALITY LOWER NIB
+
+       uint32_t centLN = ((raw_data >> 28) & 0xF);
+
+       // MHT with HF
+
+       raw_data = block.payload()[iFrame+5];
+
+       l1t::EtSum mhthf = l1t::EtSum();
+    
+       mhthf.setHwPt(raw_data & 0xFFF);
+       mhthf.setHwPhi((raw_data >> 12) & 0xFF);
+       mhthf.setType(l1t::EtSum::kMissingHtHF);       
+       mhthf.setP4( l1t::CaloTools::p4Demux(&mhthf) );
+
+       LogDebug("L1T") << "MHThf: phi " << mhthf.hwPhi() << " pT " << mhthf.hwPt() << " bx " << bx;
+
+       res_->push_back(bx,mhthf);
+
+       // HTHF ASYMM
+       
+       l1t::EtSum htHFAsym = l1t::EtSum();
+       htHFAsym.setHwPt( (raw_data>>20) & 0xFF );
+       htHFAsym.setType( l1t::EtSum::kAsymHtHF );
+
+       res_->push_back(bx, htHFAsym);
+
+
+       // CENTRALITY UPPER NIB
+
+       uint32_t centUN = ((raw_data >> 28) & 0xF);
+
+       l1t::EtSum cent = l1t::EtSum();
+       cent.setHwPt(centUN & centLN); 
```

Comment:
Hi @bundocka 
Is this bitwise AND intended here? Could this operation not be done in the FW already?